### ISSUE TITLE
Animation Timeline Crash

### ISSUE THREAD

**Issue by DW251** [Scirra/Construct-3-bugs#5911]
## Problem description

While using a Timeline, Construct 3 crashes. 

## Attach a .c3p

https://wetransfer.com/downloads/adfce0d46a8b2a894f70e59f6557ace120220721093100/407596e4a4686e23e2d743a82742346920220721093214/ab8ec7?utm_campaign=WT_email_tracking&utm_content=general&utm_medium=download_button&utm_source=notify_recipient_email

## Steps to reproduce

1. Create a timeline
2. Choose an object and animate it by moving it's Y position by around 160 pixels. There will be two keyframes in total, animation time total is around 1 second. 
3. When playing the animation in the editor, after playing the animation Construct will crash. Usually when clicking off of the timeline to select something else in the layout.

## Observed result

After playing the animation Construct will crash. Usually when clicking off of the timeline to select something else in the layout.

## Expected result

Not crashing, just moving to the layout.

## More details
I have only experienced this on Chrome, but I haven't tried any others.

I haven't been using Construct 3 for long, so have no idea when this crash started happening.

## System details

<details><summary>View details</summary>

PASTE HERE
Error report information
Type: unhandled exception
File: https://editor.construct.net/r302/components/bars/timelineBar/timelineBar.js, line 111, col 246
Message: Uncaught TypeError: Cannot read properties of undefined (reading 'style')
Stack: TypeError: Cannot read properties of undefined (reading 'style') at $1.ymc (https://editor.construct.net/r302/components/bars/timelineBar/timelineBar.js:111:246) at $1.Khb (https://editor.construct.net/r302/components/bars/timelineBar/timelineBar.js:104:63) at $1.jGa (https://editor.construct.net/r302/components/bars/timelineBar/timelineBar.js:105:175) at https://editor.construct.net/r302/components/bars/timelineBar/timelineBar.js:105:221
Construct version: r302
URL: https://editor.construct.net/
Date: Wed Jul 20 2022 15:45:59 GMT+0100 (British Summer Time)
Uptime: 718 s
Platform information
Product: Construct 3 r302 (stable)
Browser: Chrome 103.0.5060.64
Browser engine: Chromium
Context: browser
Operating system: Chrome OS 14816.82.0
Device type: desktop
Device pixel ratio: 1.25
Logical CPU cores: 2
Approx. device memory: 4 GB
User agent: Mozilla/5.0 (X11; CrOS x86_64 14816.82.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36
Language setting: en-US
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: AMD
Renderer: AMD STONEY (DRM 3.42.0, 5.10.114-16024-gbdf1547bd4f4, LLVM 12.0.1)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 2048
Extensions: EXT_color_buffer_float, EXT_color_buffer_half_float, EXT_disjoint_timer_query_webgl2, EXT_float_blend, EXT_texture_compression_bptc, EXT_texture_compression_rgtc, EXT_texture_filter_anisotropic, EXT_texture_norm16, OES_texture_float_linear, WEBGL_compressed_texture_astc, WEBGL_compressed_texture_etc, WEBGL_compressed_texture_etc1, WEBGL_compressed_texture_s3tc, WEBGL_compressed_texture_s3tc_srgb, WEBGL_debug_renderer_info, WEBGL_debug_shaders, WEBGL_draw_instanced_base_vertex_base_instance, WEBGL_lose_context, WEBGL_multi_draw, WEBGL_multi_draw_instanced_base_vertex_base_instance, WEBGL_video_texture, WEBGL_webcodecs_video_frame

</details>

**Comment by DiegoScirra** [comment]
Can you provide more detailed steps to reproduce the crash? I tried doing what you described, but I can't see any problem.

**Comment by F3der1co** [comment]
I don't know if this is the same here, but I also had timeline related crashes (where keyframes where offset from the step markers, and using next master keyframe gets stuck)
Sadly couldn't find a way to reproduce it, especially as saving the unstable timeline state before the crash and reopening the project seemed to fix it.
Not sure if this is related.

**Comment by DW251** [comment]
@DiegoScirra hi there, basically the time line editor crashes a lot when I try and use it. It is sometimes it happens when I move a key frame, but sometimes it happens when I preview the animation I've just made key frames for and then I click off the time line. Sometimes happens when I press the Space key on the timeline to pause the preview, and a habit from using DAWs and timeline based software such as Adobe Premiere.

The object I animated also had movement behaviours such as bullet etc on it. The preview would still play, ignoring those behaviours.

To summarise, I have only encountered crashes when using the timeline to animate an object moving. I have only encountered crashing when using the timeline. Love the software though :)

**Comment by DiegoScirra** [comment]
@F3der1co I did fix the issue with the play head getting stuck on keyframes outside of the current step.

Unfortunately I am going to have to close this issue as there isn't anything else really useful. Will continue looking for problems.
